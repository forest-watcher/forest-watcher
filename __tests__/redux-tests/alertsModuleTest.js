import 'react-native';
import alertsReducer, { getAreaAlerts } from 'redux-modules/alerts';
// todo test realm actions: resetAlertsDb & saveAlertsToDb ?

describe('Redux Alerts Module', () => {
  it('Initial reducer state', () => {
    expect(alertsReducer(undefined, { type: 'NONE' })).toMatchSnapshot();
  });

  describe('Simple Actions: snapshot and reducer test', () => {
    // Snapshot tests the action object and the reducer state
    function simpleActionTest(action) {
      expect(action).toMatchSnapshot();
      expect(alertsReducer(undefined, action)).toMatchSnapshot();
    }

    it('getAreaAlertsWithConfidence', () => {
      const mockArea = { id: 'areaID', geostore: { id: 'geostoreID' } };
      simpleActionTest(
        getAreaAlerts(
          mockArea,
          'datasetSlugMock',
          {
            query: {
              confidenceKey: 'confidenceKeyMock',
              datastoreId: 'datasetSlugMock',
              dateKey: 'dateKeyMock',
              minDateKey: 'minDateKeyMock',
              tableName: 'tableNameMock'
            }
          },
          new Date(Date.UTC(2021, 4, 25, 12, 0, 0))
        )
      );
    })

    it('getAreaAlerts', () => {
      const mockArea = { id: 'areaID', geostore: { id: 'geostoreID' } };
      simpleActionTest(
        getAreaAlerts(
          mockArea,
          'datasetSlugMock',
          {
            query: {
              datastoreId: 'datasetSlugMock',
              dateKey: 'dateKeyMock',
              minDateKey: 'minDateKeyMock',
              tableName: 'tableNameMock'
            }
          },
          new Date(Date.UTC(2021, 4, 25, 12, 0, 0))
        )
      );
    });
  });
});
