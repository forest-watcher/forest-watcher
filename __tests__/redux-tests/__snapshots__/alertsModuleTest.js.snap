// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux Alerts Module Initial reducer state 1`] = `
Object {
  "cache": Object {},
  "queue": Array [],
  "syncError": false,
}
`;

exports[`Redux Alerts Module Simple Actions: snapshot and reducer test getAreaAlerts 1`] = `
Object {
  "meta": Object {
    "offline": Object {
      "commit": Object {
        "meta": Object {
          "alertId": "areaID_datasetSlugMock",
          "area": Object {
            "geostore": Object {
              "id": "geostoreID",
            },
            "id": "areaID",
          },
          "confidenceKey": undefined,
          "datasetSlug": "datasetSlugMock",
          "dateKey": "dateKeyMock",
          "minDate": 2021-05-25T12:00:00.000Z,
        },
        "type": "alerts/GET_ALERTS_COMMIT",
      },
      "effect": Object {
        "deserialize": false,
        "headers": Object {
          "Content-Type": "text/csv",
          "Origin": "com.wri.forestwatcher",
          "x-api-key": undefined,
        },
        "url": "undefined/dataset/undefined/latest/query/csv?geostore_origin=rw&geostore_id=geostoreID&sql=select latitude, longitude, dateKeyMock from tableNameMock where dateKeyMock > '2021-05-25' ORDER BY dateKeyMock DESC LIMIT 10000",
      },
      "rollback": Object {
        "meta": Object {
          "alertId": "areaID_datasetSlugMock",
        },
        "type": "alerts/GET_ALERTS_ROLLBACK",
      },
    },
  },
  "payload": "areaID_datasetSlugMock",
  "type": "alerts/GET_ALERTS_REQUEST",
}
`;

exports[`Redux Alerts Module Simple Actions: snapshot and reducer test getAreaAlerts 2`] = `
Object {
  "cache": Object {},
  "queue": Array [
    "areaID_datasetSlugMock",
  ],
  "syncError": false,
}
`;

exports[`Redux Alerts Module Simple Actions: snapshot and reducer test getAreaAlertsWithConfidence 1`] = `
Object {
  "meta": Object {
    "offline": Object {
      "commit": Object {
        "meta": Object {
          "alertId": "areaID_datasetSlugMock",
          "area": Object {
            "geostore": Object {
              "id": "geostoreID",
            },
            "id": "areaID",
          },
          "confidenceKey": "confidenceKeyMock",
          "datasetSlug": "datasetSlugMock",
          "dateKey": "dateKeyMock",
          "minDate": 2021-05-25T12:00:00.000Z,
        },
        "type": "alerts/GET_ALERTS_COMMIT",
      },
      "effect": Object {
        "deserialize": false,
        "headers": Object {
          "Content-Type": "text/csv",
          "Origin": "com.wri.forestwatcher",
          "x-api-key": undefined,
        },
        "url": "undefined/dataset/undefined/latest/query/csv?geostore_origin=rw&geostore_id=geostoreID&sql=select latitude, longitude, dateKeyMock, confidenceKeyMock from tableNameMock where dateKeyMock > '2021-05-25' ORDER BY dateKeyMock DESC LIMIT 10000",
      },
      "rollback": Object {
        "meta": Object {
          "alertId": "areaID_datasetSlugMock",
        },
        "type": "alerts/GET_ALERTS_ROLLBACK",
      },
    },
  },
  "payload": "areaID_datasetSlugMock",
  "type": "alerts/GET_ALERTS_REQUEST",
}
`;

exports[`Redux Alerts Module Simple Actions: snapshot and reducer test getAreaAlertsWithConfidence 2`] = `
Object {
  "cache": Object {},
  "queue": Array [
    "areaID_datasetSlugMock",
  ],
  "syncError": false,
}
`;
